### PR TITLE
Fix Sapi4 pitch change #12311

### PR DESCRIPTION
**Pull request by cary-rowen**
### Link to issue number:
#12311 

### Summary of the issue:
This pr solves the problem in #12311 , To put it simply, Starting with version 2019.3, the "Capital pitch change" when using the SAPI4 synthesizer was invalid.
But this must be incorrect behavior.
For visually impaired developers, this feature is especially important when typing and checking code.

### Description of how this pull request fixes the issue,
I don't know why.
this function is marked as unavailable in the speak function, I managed to add this part of the code, and tested it, it works well.

### Testing strategy:

1. Start NVDA and press NVDA + CTRL + S to select "Microsoft Speech API version 4" and confirm.
2. At this point the NVDA should have switched to the SAPI4 synthesizer and you will need to press NVDA + CTRL + V to open the Voice Setup dialog.
3. Use the Tab key to navigate until you hear "Capital pitch change percentageeditselected 30" You can change this value to 40 (or higher, or leave it unchanged).
4. Then use Windows + R to open the Run dialog and type notepad and confirm, this will open notepad allowing you to type some text.
5. Press the Caps lock key (it may be located to the left of the letter A) You may need to press it twice in a row until you hear "Caps lock on".
6. Try to enter some letters, such as the letter "C" (or "D" or something else)
7. Use Left Arrow or Right Arrow to move between these letters.

**You will hear NVDA report these capital letters in a higher pitch.**

### Known issues with pull request:
None

### Change log entries:
Bug fixes: 
- Fixed the pitch change failure in Sapi4 speech synthesizer. (#12311)

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual tests.
- [ ] User Documentation.
- [x] Change log entry.
- [ ] Context sensitive help for GUI changes.

### CODE REVIEW

File: source/synthDrivers/sapi4.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 	VOICECHARSET
 )
 import config
+import speech
```
Comment (lukaszgo1) [inline source/synthDrivers/sapi4.py:38]: You should not rely on `PitchCommand` being imported into speech. Please import it from `speech.commands`.

File: source/synthDrivers/sapi4.py [hunk 2]
```diff
@@ -67,7 +68,7 @@ def IUnknown_Release(self, this, *args, **kwargs):
 class SynthDriver(SynthDriver):
 
 	name="sapi4"
-	description="Microsoft Speech API version 4"
+	description="NewSpeech 4"
```
Comment (lukaszgo1) [inline source/synthDrivers/sapi4.py:71]: Why this change?

**Review by lukaszgo1 — CHANGES_REQUESTED** [review]
This should be converted into a draft until Linter warnings are fixed.

File: source/synthDrivers/sapi4.py (hunk 2 above)
Comment (cary-rowen) [reply source/synthDrivers/sapi4.py]: I'm very sorry, this is my mistake, it has been changed back to the original description by me

File: source/synthDrivers/sapi4.py (hunk 1 above)
Comment (cary-rowen) [reply source/synthDrivers/sapi4.py]: Thank you, I have made the changes according to your prompts.

**Comment by cary-rowen** [comment]
Hi!
How should I fix Linter warning: "E225 missing whitespace around operator"
Other codes in NVDA don't seem to add spaces before and after the operators. Sorry, I may not be familiar with github.
Thanks.
@lukaszgo1

**Comment by lukaszgo1** [comment]
> 
> 
> How should I fix Linter warning: "E225 missing whitespace around operator"
> Other codes in NVDA don't seem to add spaces before and after the operators. Sorry, I may not be familiar with github.


For the new code there are spaces on both sides of operators so you just need to add them. The code is being updated gradually to the new style that is why it may seems inconsistent.

**Comment by XLTechie** [comment]
Specifically, various operators should look like this:

result = 1 + 1

def test(a, b, c=None):

See the linting readme, at 
[tests/lint/readme.md](https://github.com/nvaccess/nvda/tree/master/tests/lint)

**Comment by cary-rowen** [comment]
I corrected all Linter warnings. Can the PR be reviewed now?
Thanks.
@lukaszgo1

**Review by feerrenrut — APPROVED** [review]
Thanks @cary-rowen

**Comment by cary-rowen** [comment]
I have improved some descriptions of the PR, and it is now clearer.

**Comment by cary-rowen** [comment]
Most of the work of this PR was done by my friend @chenfu2000 Currently, it looks good.

File: source/synthDrivers/sapi4.py [hunk 3]
```diff
@@ -123,6 +129,16 @@ def speak(self,speechSequence):
 			elif isinstance(item, CharacterModeCommand):
 				textList.append("\\RmS=1\\" if item.state else "\\RmS=0\\")
 				charMode=item.state
+			elif isinstance(item, PitchCommand):
+				offset = int(config.conf["speech"]['sapi4']["capPitchChange"])
```
Comment (leonardder) [inline source/synthDrivers/sapi4.py:133]: I was looking at this code and this really feels incorrect to me. It looks like the caps pitch change is applied to every pitch change, regardless of whether it is related to capital letters.

Comment (cary-rowen) [reply source/synthDrivers/sapi4.py:133]: OK, I see you've opened a PR to try and fix this, that's great.
